index.js

```diff
@@ -39,6 +39,13 @@ function Template(name, opts) {
         if (e.code !== 'MODULE_NOT_FOUND') { throw e; }
         this.exclusions = [];
     }
+    this.fromJson = false;
+    // Allowing overriding of values to be passed in instead of gathering them from user input.
+    if (opts.jsonValues.length && opts.jsonValues.length > 0) {
```

Comment:
this is not how you check for the size of an object.

Use `Object.keys(opts.jsonValues).length` instead.